### ISSUE TITLE
JCROM cannot map protected properties. 

### ISSUE THREAD

**Issue by GoogleCodeExporter** [Kobee1203/jcrom#93]
```
It appears to be impossible to map any protected property when adding a new 
Node.  Protected properties should be able to be set on adds, but not updates.  
In some cases, JCROM provides annotations for JCR protected properties, e.g 
@JcrCreated maps to jcr:created.  However, there are no annotations for other 
protected attributes such as jcr:createdBy and jcr:lockOwner.  It is impossible 
to map these fields.
    @JcrCreated
    private Date createDate; // this works

    @JcrProperty(name="jcr:createdBy")
    private String createdBy; // this doesn't  

    @JcrProperty(name="jcr:lockOwner")
    private String lockOwner; // this doesn't

jcr:createdBy is protected, as it should be.  When trying to add this Node, a 
JcrMappingException is thrown.
javax.jcr.nodetype.ConstraintViolationException: Unable to perform operation. 
Node is protected.

At the very minimum, annotations should be provided to map all protected JCR 
standard properties, similar to @JcrCreated.  However, this would not be a 
complete solution.  Suppose I define a custom type.

[my:sample] > 'nt:base'
- my:immutableField (string) mandatory protected version

This does not work, unless the protected constraint is removed.  Of course, one 
can enforce immutability in the Java code, but it would be nice if the 
repository could also enforce it.  

In short:  allow protected properties to be set on Node creation, but not on 
Node update.  
```

Original issue reported on code.google.com by `douglass...@me.com` on 12 Oct 2012 at 5:39
